clients/src/main/java/org/apache/kafka/clients/producer/ProducerRecord.java

```diff
@@ -38,11 +38,14 @@
  */
 public class ProducerRecord<K, V> {
 
+    public static final Long DEFAULT_EXPECTED_OFFSET = -1L;
```

Comment:
personally, every use of `Optional` I've seen in the product is _missing_ the greater point (i.e. something called monadic composition / chaining) of when `Optional` shines. So, in general, I recommend never using `Optional`. 

See https://dzone.com/articles/whats-wrong-java-8-part-iv